### PR TITLE
feat: add endpoint to handle re-orgs

### PR DESCRIPTION
**Pull request by juanmahidalgo**
Given a `last_valid_block` and a `schema`, the endpoint deletes all the rows with `block_number` > than the last valid one on all the tables with `block_number` in the `schema` specified.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 dist/
 node_modules/
 **/.DS_Store
-coverage
\ No newline at end of file
+coverage
+.env
```
Comment (cyaiox) [inline .gitignore:5]: missing EOL

File: src/types.ts [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import type { IFetchComponent } from "@well-known-components/http-server"
+import type { IPgComponent } from "@well-known-components/pg-component"
+import type * as authorizationMiddleware from "decentraland-crypto-middleware"
```
Comment (cyaiox) [inline src/types.ts:3]: is it required to import all the types? or could it work importing just: `DecentralandSignatureContext`?

File: src/logic/reorg/component.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          .append(SQL`.`.append(table_name).append(SQL` WHERE block_number::numeric > ${blockNumber};`))
+
+        await client.query(deleteQuery)
+      }
+
+      await client.query("COMMIT")
+
+      return Promise.resolve()
+    } catch (e) {
+      console.error(e)
```
Comment (cyaiox) [inline src/logic/reorg/component.ts:39]: should be here an `await client.query("ROLLBACK")`?

Comment (juanmahidalgo) [reply src/logic/reorg/component.ts:39]: absolutely right! added!

File: src/types.ts (hunk 2 above)
Comment (juanmahidalgo) [reply src/types.ts]: updated to import only `DecentralandSignatureContext`!

**Review by cyaiox — APPROVED** [review]
🚀

**Pull request merged**